pkg/istio/client.go

```diff
@@ -63,30 +67,59 @@ func (c *Client) getVirtualService(ctx context.Context) (*istioclientapi.Virtual
 	return virtualService, nil
 }
 
-func (c *Client) createVirtualService(ctx context.Context, watcher *v1alpha1.Watcher,
+func (c *Client) CreateVirtualService(ctx context.Context, watcher *v1alpha1.Watcher,
 ) (*istioclientapi.VirtualService, error) {
 	if watcher == nil {
 		return &istioclientapi.VirtualService{}, nil
 	}
-	_, err := c.NetworkingV1beta1().
-		Gateways(metav1.NamespaceDefault).
-		Get(ctx, c.gatewayName, metav1.GetOptions{})
+
+	gateway, err := c.lookupGateway(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("error getting configured istio gateway: %w", err)
+		return nil, err
 	}
+
 	virtualSvc := &istioclientapi.VirtualService{}
 	virtualSvc.SetName(c.virtualServiceName)
 	virtualSvc.SetNamespace(metav1.NamespaceDefault)
-	virtualSvc.Spec.Gateways = append(virtualSvc.Spec.Gateways, c.gatewayName)
+	if gateway != nil {
+		gwKey := apitypes.NamespacedName{Namespace: gateway.Namespace, Name: gateway.Name}
+		virtualSvc.Spec.Gateways = append(virtualSvc.Spec.Gateways, gwKey.String())
+	}
 	virtualSvc.Spec.Hosts = append(virtualSvc.Spec.Hosts, "*")
 	virtualSvc.Spec.Http = []*istioapi.HTTPRoute{
 		prepareIstioHTTPRouteForCR(watcher),
 	}
+
 	return c.NetworkingV1beta1().
 		VirtualServices(metav1.NamespaceDefault).
 		Create(ctx, virtualSvc, metav1.CreateOptions{})
 }
 
+func (c *Client) lookupGateway(ctx context.Context) (*istioclientapi.Gateway, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: c.gatewaySelector,
+	}
+	gateways, err := c.NetworkingV1beta1().
+		Gateways(metav1.NamespaceAll).
+		List(ctx, lo)
+
+	if err != nil {
+		return nil, fmt.Errorf("error looking up Istio gateway with label %q: %w", c.gatewaySelector, err)
+	}
+
+	if len(gateways.Items) == 0 {
+		c.logger.Info("Warning: No matching Istio gateways found", "labelSelector", c.gatewaySelector)
+		return nil, nil
```

Comment:
why is this an info warning and not an error for lookup?

Reply:
This comes from the issue Acceptance Criteria. @khlifi411 could you confirm?

Reply:
Actually, the ACs were updated by @jakobmoellersap [here](https://github.com/kyma-project/lifecycle-manager/pull/231#discussion_r987619930) 😅
@jakobmoellersap would be great if you adjust the ACs 👍

Reply:
The ACs are correct. However imo, the proper handling here should return the error and send a Kubernetes event to the resource with the warning. This is also what I meant in the ACs. Simply logging an error in the controller is not visible to most users deploying the controller. They will only look at the CR. If you call `logger.Info` it will only be visible in controller-logs. The ACs are still fitting imo.

Reply:
Agree in this point with @jakobmoellersap

Reply:
@jakobmoellersap I agree about your description above 👍 
I still think it's unclear (at least for me) in the ACs if `output a warning` means `fire a warning event` or `log a warning`.

Reply:
@khlifi411 @jakobmoellersap  Same for me, I understood this AC as "continue processing instead of firing an error, but log the warning".